src/main/scala/com/gu/zuora/crediter/CreditTransferService.scala

```diff
@@ -82,8 +84,6 @@ class CreditTransferService(command: CreateCreditBalanceAdjustmentCommand)(impli
         maybeId.foreach(id => logger.info(s"Successfully created Credit Balance Adjustment, ID: $id"))
         maybeId
       }).flatten
```

Comment:
if you are relying on side effects, I wouldn't leave things lazy, it would be better to toList it before leaving the function